### PR TITLE
[For review] Use Boto3 instead of Bash

### PR DESCRIPTION
**Pull request by waisbrot**
This isn't meant as a proper PR, just a something to take a look at. If you like it, I'd be happy to do some clean-up work of course (it's pretty sloppy right now).

This replaces the Bash script with Python. The main reasoning behind that is that I find Python easier, and we were running a bunch of Python anyway (in the form of the `aws` tool).

The other thing it does is create a service record in DNS. I can then configure etcd proxies like this:

```
  etcd2:
    name: worker-$private_ipv4
    proxy: on
    listen-client-urls: http://localhost:2379
    discovery-srv: "whoop.internal"
```

Thanks for bumping the other PR @placydo !

### CODE REVIEW

**Comment by placydo** [comment]
Thanks for sharing Your code!

File: cluster.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+CLIENT_PORT = int(os.getenv('CLIENT_PORT', 2379))
+PEER_PORT = int(os.getenv('PEER_PORT', 2380))
+
+MAX_RETRIES = int(os.getenv('MAX_RETRIES', 10))
+
+HOSTED_ZONE_ID = os.getenv('HOSTED_ZONE_ID', None)
+DOMAIN_NAME = os.getenv('DOMAIN_NAME', None)
+
+boto = None
```
Comment (placydo) [inline cluster.py:28]: Want to hear Your opinion. For me global variable isn't solution. Maybe object creation should be handled by constructor and boto could be passed as argument?

File: cluster.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        Given a list of IP addresses, see if any of them will tell us about an existing
+        cluster. Update the object with the results of the hunt.
+        """
+        for peer in self.candidates:
+            client_url = peer.client_url
+            LOG.debug('hunting for a Etcd cluster: %s', client_url)
+            try:
+                response = requests.get('{}/v2/members'.format(client_url), timeout=3).json()
+                found_members = response['members']
+                LOG.debug('found a cluster of %s at %s: %s', len(found_members), client_url, found_members)
```
Comment (placydo) [inline cluster.py:141]: Maybe all logged text should start with capital letter?

File: cluster.py [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+        self._hunt_for_cluster()
+
+    def _hunt_for_cluster(self):
+        """
+        Given a list of IP addresses, see if any of them will tell us about an existing
+        cluster. Update the object with the results of the hunt.
+        """
+        for peer in self.candidates:
+            client_url = peer.client_url
+            LOG.debug('hunting for a Etcd cluster: %s', client_url)
```
Comment (placydo) [inline cluster.py:137]: Maybe all logged text should start with capital letter?

File: cluster.py (hunk 2 above)
Comment (placydo) [reply cluster.py:141]: Also not sure if You should remove bash script. This will allow to phase out bash script rather than forcing removal.

**Comment by tjcorr** [comment]
In general I like the direction you are headed. I'm by no means a Python expert so give me a bit to dig through this and test it out.

**Comment by tjcorr** [comment]
I stood up a quick cluster with the current code and the most basic stuff works.

**Comment by waisbrot** [comment]
Closing this to tidy up my PR history.

**Pull request closed without merging**